Code/Yospace/Source/YospaceID3MetadataHandler.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  YospaceID3MetadataHandler.swift
+//
+//  Created by Raffi on 30/05/2024.
+//  Copyright © 2024 THEOplayer. All rights reserved.
+//
+
+import THEOplayerSDK
+import YOAdManagement
+
+class YospaceID3MetadataHandler {
+    private let player: THEOplayerSDK.THEOplayer
+    private let session: YOSession
+    private var addTrackEventListener: THEOplayerSDK.EventListener?
+    private var addCueEventListener: THEOplayerSDK.EventListener?
+    private var metadata: YospaceTimedMetadata = .init()
+
+    init(player: THEOplayerSDK.THEOplayer, session: YOSession) {
+        self.player = player
+        self.session = session
+        self.addTrackEventListener = self.player.textTracks.addEventListener(type: THEOplayerSDK.TextTrackListEventTypes.ADD_TRACK) { [weak self] event in
+            self?.addTrackEventHandler(event: event)
+        }
+    }
+
+    private func addTrackEventHandler(event: THEOplayerSDK.AddTrackEvent) {
+        if let textTrack: THEOplayerSDK.TextTrack = event.track as? THEOplayerSDK.TextTrack,
+           textTrack.kind == THEOplayerSDK.TextTrackKind.metadata._rawValue,
+           textTrack.type == "id3" {
+            self.addCueEventListener = textTrack.addEventListener(type: THEOplayerSDK.TextTrackEventTypes.ADD_CUE) { [weak self] event in
+                self?.addCueEventHandler(event: event)
+            }
+        }
+    }
+
+    private func addCueEventHandler(event: THEOplayerSDK.AddCueEvent) {
```

Comment:
We shouldn't be calling `timedMetadataWasCollected` from within an `addcue` listener. [As per Yospace docs](https://developer.yospace.com/sdk-documentation/apple/userguide/latest/en/provide-necessary-information-to-the-sdk.html) ("3.5.1.2.2.2. Step 2: Forward the Timed Metadata to the SDK"), we must call it only when the *playback position reaches that cue*.

We should be using the `entercue` or `cuechange` event instead. See [TimedMetadataHandler.kt](https://github.com/THEOplayer/android-connector/pull/20/files#diff-1e16e768d1d0b6032bbcc1a802d8c11f22c9834bd2206e6adeaa272010cc1256) in the Android connector.

Reply:
Done